src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        currentState = { ...currentState };
```

Comment:
There's no need to create a new copy of `currentState` on line 21 since you're not changing the structure of the object, but rather deleting properties from it.

Reply:
Fix this comment